front-end/src/organisms/WorkspaceSettings.js

```diff
@@ -1,63 +1,81 @@
 import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
 
-import { ExitIcon, AddCollab } from "../atoms/icons";
+import { ExitIcon, AddCollabIcon } from "../atoms/icons";
+import ErrorMessage from "../atoms/ErrorMessage";
 import CollaboratorItem from "../molecules/CollaboratorItem";
 import Button from "../atoms/Button";
 import "./WorkspaceSettings.css";
 
+import { isValidEmail } from "../util/util";
 
-function WorkspaceSettings({onChangeVisibility}) {
+
+function WorkspaceSettings({ onChangeVisibility }) {
   const componentName = "WorkspaceSettings";
 
   const [renderedName, setRenderedName] = useState("");
   const [renderedCollaborators, setRenderedCollaborators] = useState([]);
   const [renderSave, setRenderSave] = useState(false);
+  const [error, setError] = useState("");
 
   // formats a list of strings to a list of object contains {email, pending}
-  const formatCollaborators = (list) =>
-    list.map((i) => ({email: i, pending: false}));
+  const formatCollaborators = (list) => list.map((i) => ({ email: i, pending: false }));
 
-  const handleCollaboratorSubmit = (e) => {
-    if (e.key === "Enter") {
+  const handleKeyPress = e => {
```

Comment:
I'm not sure if `handleKeyPress` method is the best name for this, this can mean when any key is pressed.

What about `handleCollaboratorSubmitOnEnter`.